Rework particle documentation

slfuchs opened the pull request:
<!--
* Title: Provide a general summary of your changes in the Title above.

* Assignees:  If you know anyone who should likely handle bringing this pull request (PR) to completion -- that's usually yourself -- select them from the Assignees drop-down on the right.

* Labels: Update the label of the issue(s) addressed by this pull request to "Under Review".
-->

## Description and Context
<!--
Provide a brief and concise description of your proposed change. Questions you should think about:
* Why is this change required?  What problem does it solve?
* Is there a bigger picture? Is this PR a part of a larger set of changes? Which further steps are planned after merging this PR, if any?
* How has the proposed implementation been verified and tested?

Keep the description of the PR always up-to-date and concise.
-->

Rework the particle related documentation mainly adding some corrections.

## Related Issues and Merge Requests
<!--
If applicable, let us know how this pull request is related to any other open issues or pull requests by linking to them here.
Some suggestion for keywords:
Closes (will automatically close mentioned issue if merged), Blocks, Related to
-->

sebproell commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -159,20 +159,19 @@ Particle Discretization
     TYPE <typestring> POS <x> <y> <z> [RAD <radius>]
 
-The type provides a particle set, to which a material can be assigned in the section :ref:`PARTICLE DYNAMIC <SECparticledynamic>`,
+The type keyword provides a particle phase, to which a material can be assigned in the section :ref:`PARTICLE DYNAMIC <SECparticledynamic>`,
```
```suggestion
The ``TYPE`` keyword provides a particle phase, to which a material can be assigned in the section :ref:`PARTICLE DYNAMIC <SECparticledynamic>`,
```

sebproell commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -159,20 +159,19 @@ Particle Discretization
 Currently, two different particle methods are implemented: DEM (discrete element method) and SPH (smoothed particle hydrodynamics).
-For both methods, particles are simply defined by its position and optionally by its radius (only available for DEM),
-the particle number may not be entered.
+For both methods, particles are simply defined by its spatial position and optionally by its radius (only available for DEM).
```
```suggestion
For both methods, particles are simply defined by their spatial position and optionally by their radius (only available for DEM).
```

sebproell commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -181,16 +180,23 @@ For SPH, the following different phases may be defined:
-The radius of the particles are by default defined in the material definition. For SPH, this is the only option available.
-For DEM, however, as mentioned above, they can also be defined per particle.
-For this definition option, one has to define the option :ref:`INITIAL_RADIUS <particledynamic_dem_initial_radius>` in :ref:`PARTICLE DYNAMIC <SECparticledynamic_dem>`::
+Note that with SPH the particles are allowed to change its phase during a simulation based on the input parameter `PHASECHANGETYPE` and `PHASECHANGEDEFINITION` in the section :ref:`PARTICLE DYNAMIC/SPH <SECparticledynamic_sph>`.
```
Maybe @ischeider can confirm this: I think we should use double backticks, right?

```suggestion
Note that with SPH the particles are allowed to change its phase during a simulation based on the input parameter ``PHASECHANGETYPE`` and ``PHASECHANGEDEFINITION`` in the section :ref:`PARTICLE DYNAMIC/SPH <SECparticledynamic_sph>`.
```

slfuchs replied:
You are right, it should be double backticks.

ppraegla reviewed (dismissed):
LGTM! Thanks, @slfuchs

slfuchs commented:
@sebproell Accepting your proposed changes created three separate commits. Is there a way to accept all changes at once to get it within one commit? Otherwise I would manually change it and push one single commit, to keep the git history clean.

mayrmt commented:
@slfuchs , feel free to also include references to (your) papers, if they are relevant for the implemented methods.

sebproell commented:
@slfuchs We squash the commits manually. We haven't activated the Github feature to squash on merge, because this does not (always) create a merge commit but just applies your commit.

ischeider commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -181,16 +180,23 @@ For SPH, the following different phases may be defined:
-The radius of the particles are by default defined in the material definition. For SPH, this is the only option available.
-For DEM, however, as mentioned above, they can also be defined per particle.
-For this definition option, one has to define the option :ref:`INITIAL_RADIUS <particledynamic_dem_initial_radius>` in :ref:`PARTICLE DYNAMIC <SECparticledynamic_dem>`::
+Note that with SPH the particles are allowed to change its phase during a simulation based on the input parameter ``PHASECHANGETYPE`` and ``PHASECHANGEDEFINITION`` in the section :ref:`PARTICLE DYNAMIC/SPH <SECparticledynamic_sph>`.
```
```suggestion
Note that with SPH the particles are allowed to change their phase during a simulation based on the input parameter ``PHASECHANGETYPE`` and ``PHASECHANGEDEFINITION`` in the section :ref:`PARTICLE DYNAMIC/SPH <SECparticledynamic_sph>`.
```

ischeider commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -181,16 +180,23 @@ For SPH, the following different phases may be defined:
-The parameter :math:`mu` is defined by the initial radius, i.e., in the material definition,
-while the parameter :math:`\sigma` is defined by the parameter :ref:`RADIUSDISTRIBUTION_SIGMA <particledynamic_dem_radiusdistribution_sigma>` in :ref:`PARTICLE DYNAMIC <SECparticledynamic_dem>`::
+Additionally, it is possible to define a normal or a log-normal distribution of the radius, defined by two parameters, :math:`\mu` and :math:`\sigma`.
+The parameter :math:`mu` is defined by the initial radius in the material definition,
```
```suggestion
The parameter :math:`\mu` is defined by the initial radius in the material definition,
```

ischeider commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -181,16 +180,23 @@ For SPH, the following different phases may be defined:
-Additionally, it is possible to define a normal or a log-normal distribution, defined by two parameters, :math:`\mu` and :math:`\sigma`.
-The parameter :math:`mu` is defined by the initial radius, i.e., in the material definition,
-while the parameter :math:`\sigma` is defined by the parameter :ref:`RADIUSDISTRIBUTION_SIGMA <particledynamic_dem_radiusdistribution_sigma>` in :ref:`PARTICLE DYNAMIC <SECparticledynamic_dem>`::
+Additionally, it is possible to define a normal or a log-normal distribution of the radius, defined by two parameters, :math:`\mu` and :math:`\sigma`.
```
Actually, what does a size distribution mean? Is the radius chosen randomly within the distribution? I am not a particle expert, but maybe it is worth including a more in-depth explanation.

slfuchs replied:
Typically in powder spreading applications the individual powder grains are not of equal size, i.e., with given fixed radius. Accordingly, based on a characterization of the powder the size of particles can be described by a distribution (normal or log-normal) with mean and standard deviation. @ppraegla can probably give you more information or also link one of the papers from our group. In the implementation, the radius of each particle is at the beginning of a simulation randomly set based on the selected distribution.

ischeider reviewed (commented):
Thank you very much for improving the particle documentation, which was initially written by a non-expert ;-)

slfuchs commented:
> Thank you very much for improving the particle documentation, which was initially written by a non-expert ;-)

Thanks for your initial effort!

slfuchs commented:
@sebproell I included your and @ischeider suggestions and squashed the commits. From my side this PR is ready for the next round of reviews.

> @slfuchs , feel free to also include references to (your) papers, if they are relevant for the implemented methods.

@mayrmt As of now I leave it as it is. But I think about extending the documentation in the future and add reference to our publications then.

sebproell commented on doc/readthedocs/geometrydefinition.rst:
```diff
@@ -159,20 +159,19 @@ Particle Discretization
-For both methods, particles are simply defined by its position and optionally by its radius (only available for DEM),
-the particle number may not be entered.
+For both methods, particles are simply defined by their spatial position and optionally by their radius (only available for DEM).
+A unique global id is assigend to each particle automatically during runtime.
```
```suggestion
A unique global ID is assigned to each particle automatically during runtime.
```

sebproell commented on doc/readthedocs/problemtypes.rst:
```diff
@@ -161,14 +161,16 @@ Particle
-This is a completely different problem type, which comes along without elements.
-Instead, the method is based on SPH, which means *smoothed particle hydrodynamics*,
-and the particles are defined by their coordinates.
+This is a completely different discretization type that comes along without elements but uses particles, i.e., spatial discretization points.
```
```suggestion
This is a completely different discretization type without the typical mesh consisting of elements.
Instead, this mesh-free approach uses particles that can move in space.
```

I think this is what you try to say by "spatial discretization points"?

slfuchs replied:
I prefer to leave the description as I wrote it.

slfuchs replied:
Edit: I changed the description to be more precise.

sebproell reviewed (commented):
Sorry, missed these.

slfuchs commented:
@sebproell I added the new suggestions and rewrote one sentence. Let's keep the review on a level that also fits for old BACI developers that invest there leisure time here ;-)
